src/index.ts

```diff
@@ -492,18 +492,25 @@ export class Lrud {
       return this.digDown(_findChildWithClosestIndex(node, this.currentFocusNodeIndex, this.currentFocusNodeIndexRange), direction);
     }
 
+    if (!isNodeFocusable(node) && !this.doesNodeHaveFocusableChildren(node)) {
+      const parentNode = this.getNode(node.parent);
+      parentNode.activeChild = node.id
+      const nextSiblingFromNode = this.getNextChildInDirection(parentNode, direction);
+      // if the next sibling is ME, we're in an infinite loop - just return null
+      if (nextSiblingFromNode.id === node.id) {
+        return null
+      }
+      return this.digDown(nextSiblingFromNode, direction)
+    }
+
     // if we dont have an active child, use the first child
     if (!node.activeChild) {
       node.activeChild = this.getNodeFirstChild(node).id
     }
 
-    const activeChild = node.children[node.activeChild]
-
-    if (isNodeFocusable(activeChild)) {
-      return activeChild
-    }
+    let nextChild = node.children[node.activeChild]
```

Comment:
What happens if there is no first child in this instance?

Reply:
Ah think this in answered in the next line!

Reply:
Haha, yep - the next line is a clause around "if `nextChild` is not focusable, dig down again from `nextChild`"

And `digDown()` itself takes care of an invalid node in that case.